Content.Client/Light/Visualizers/PoweredLightVisualizerSystem.cs

```diff
@@ -0,0 +1,147 @@
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PoweredLightVisualizerComponent, AnimationCompletedEvent>(OnAnimationCompleted);
+    }
+
+    protected override void OnAppearanceChange(EntityUid uid, PoweredLightVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+        if (!AppearanceSystem.TryGetData(uid, PoweredLightVisuals.BulbState, out PoweredLightState state, args.Component))
+            return;
+
+        switch (state)
+        {
+            case PoweredLightState.Empty:
+                args.Sprite.LayerSetState(PoweredLightLayers.Base, "empty");
+                ToggleBlinkingAnimation(uid, false, comp, ref args);
+                break;
+            case PoweredLightState.Off:
+                args.Sprite.LayerSetState(PoweredLightLayers.Base, "off");
+                ToggleBlinkingAnimation(uid, false, comp, ref args);
+                break;
+            case PoweredLightState.On:
+                if (AppearanceSystem.TryGetData(uid, PoweredLightVisuals.Blinking, out bool isBlinking, args.Component))
+                    ToggleBlinkingAnimation(uid, isBlinking, comp, ref args);
+                if (!isBlinking)
+                {
+                    args.Sprite!.LayerSetState(PoweredLightLayers.Base, "on");
+                }
+                break;
+            case PoweredLightState.Broken:
+                args.Sprite.LayerSetState(PoweredLightLayers.Base, "broken");
+                ToggleBlinkingAnimation(uid, false, comp, ref args);
+                break;
+            case PoweredLightState.Burned:
+                args.Sprite.LayerSetState(PoweredLightLayers.Base, "burn");
+                ToggleBlinkingAnimation(uid, false, comp, ref args);
+                break;
+        }
+    }
+
+    private void OnAnimationCompleted(EntityUid uid, PoweredLightVisualizerComponent comp, AnimationCompletedEvent args)
+    {
+        if(!comp.WasBlinking)
+            return;
+        if (args.Key != "blinking")
+            return;
+        AnimationSystem.Play(uid, Comp<AnimationPlayerComponent>(uid), BlinkingAnimation(comp), "blinking");
+    }
+
+
+    private void ToggleBlinkingAnimation(EntityUid uid, bool isBlinking, PoweredLightVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (isBlinking == comp.WasBlinking)
+            return;
+        comp.WasBlinking = isBlinking;
+
+        var animationPlayer = EnsureComp<AnimationPlayerComponent>(uid);
+        if (isBlinking)
+        {
+            AnimationSystem.Play(uid, animationPlayer, BlinkingAnimation(comp), "blinking");
+        }
+        else if (AnimationSystem.HasRunningAnimation(uid, animationPlayer, "blinking"))
+        {
+            AnimationSystem.Stop(uid, animationPlayer, "blinking");
+        }
+    }
```

Comment:
move the key to the component, i dont thunk `ToggleBlinkingAnimation` needs `AppearanceChangeEvent` as an argument, make `OnAnimationCompleted` use `ToggleBlinkingAnimation`